pkg/core/managers/apis/zone/zone_manager.go

```diff
@@ -23,7 +23,8 @@ type zoneManager struct {
 }
 
 func (z *zoneManager) Delete(ctx context.Context, r model.Resource, opts ...core_store.DeleteOptionsFunc) error {
-	if err := z.validator.ValidateDelete(ctx, r.GetMeta().GetName()); err != nil {
+	options := core_store.NewDeleteOptions(opts...)
+	if err := z.validator.ValidateDelete(ctx, options.Name); err != nil {
```

Comment:
Nit: seems like it would be cleaner to pass the delete options down to `VallidateDelete`. If you agree, file a follow up issue?